tests/generator-Tests/Unit-Tests/SymbolTableTests.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using MonoDroid.Generation;
+using NUnit.Framework;
+
+namespace generatortests
+{
+	[TestFixture]
+	public class SymbolTableTests
+	{
+		[Test]
+		public void FindGenericTypes ()
+		{
+			var table = new SymbolTable ();
+
+			var list = new InterfaceGen (new GenBaseSupport {
+				Name = "System.Collections.Generic.IList`1",
+				FullName = "System.Collections.Generic.IList`1",
+				JavaSimpleName = "System.Collections.Generic.IList`1"
+			});
+
+			table.AddType (list);
+
+			var dict = new InterfaceGen (new GenBaseSupport {
+				Name = "System.Collections.Generic.IDictionary`2",
+				FullName = "System.Collections.Generic.IDictionary`2",
+				JavaSimpleName = "System.Collections.Generic.IDictionary`2"
+			});
+
+			table.AddType (dict);
+
+			Assert.NotNull (table.Lookup ("System.Collections.Generic.IList<Java.Util.Locale.LanguageRange>"));
+			Assert.NotNull (table.Lookup ("System.Collections.Generic.IList<List<Java.Util.Locale.LanguageRange>>"));
```

Comment:
To ensure that the nested `<...>` logic works, we should also add:

```
Assert.NotNull (table.Lookup ("System.Collections.Generic.IList<Dictionary<string, Java.Util.Locale.LanguageRange>>"));
```

We might also want to assert that *not only* is the `table.Lookup()` not null, but *also* what the resulting value *is*.